docs/http.md

```diff
@@ -2,93 +2,46 @@
-- `body` - associative array - request body
-- `queryParams` - associative array - with  query params
-- `compression` - (defalt: true) boolean - indicating if the request should be compressed
-- `timeout` - integer - time after which request should be aborted
-
-The `HttpResponse` is a node with fields:
-- `id` - string
-- `httpStatusCode` - integer
-- `headers` - associative array
-- `requestOptions` - associative array
-- `rawData` - dynamic
-- `failureReason` - string
-
-## Define Request
-
-To create your own request create a new component that extends `Request` (`/component/http/request/Request`).
-
-You can always create a component that will aggregate the common logic of your requests and extend that component (`MyRequest` extends `MyCommonRequest` extends `Request`).
-
-`Request` has the following interface that should be extended in your `Request` derived component:
-- `initRequest` - that will be run when the request is initialized. It is recommended to get all data from other nodes here, as exchanging the data between nodes will result in rendezvous in further process. The method is executed on `render` thread.
-- `runRequest` - If you are using `HttpService`, you will invoke `httpService.fetch` here.
-- `getRequestOptions` - this needs to return response options like URL, headers, method, body, timeout.
-- `parseResponseData` - this method will parse response data, you can for example create a new node, add data there and return it. The method is executed on the `Task` thread.
-- `generateErrorData` - here you can generate your custom error data that should be thrown on request failure. The method is executed on the `Task` thread.
-
-It is up to you how you handle your data. `parseResponseData` and `generateErrorData` should contain the appriopriate handlers.
+`HttpRequest` has the following interface that should be extended in your `Request` derived component:
+<<<<<<< Updated upstream
+- `getRequestOptions(data)` - returns an object implementing the [`HttpRequest~Options`](#Request Options)
+- `parseResponse(response)` - returns a data object (e.g. node) based on HttpResponse object (`/component/http/HttpResponse.brs`) fulfilled with data; the promise returned from `createRequest()` function will be resolved with this object. The method is executed on a task thread,
+- `generateErrorData(response)` - returns custom error data object thrown on request failure; the promise returned from `createRequest()` function will be rejected with this object. The method is executed on a task thread,
+=======
+- `getRequestOptions(data)` - returns an object implementing the `HttpRequest~Options` interface with options like URL, headers, method, body, timeout
+- `parseResponse(response)` - returns a data object (e.g. node) based on HttpResponse object (`/component/http/HttpResponse.brs`) fulfilled with data; the promise returned from `createRequest()` function will be resolved with this object. The method is executed on a task thread
```

Comment:
I have not tested this, but instead of `HttpResponse object (`/component/http/HttpResponse.brs`) fulfilled` could we just link straight to the file?

```suggestion
- `parseResponse(response)` - returns a data object (e.g. node) based on [HttpResponse](/component/http/HttpResponse.brs) object fulfilled with data; the promise returned from `createRequest()` function will be resolved with this object. The method is executed on a task thread
```

Reply:
Good idea. It turned out it works this way: `[HttpResponse](../src/components/http/HttpResponse.brs)`